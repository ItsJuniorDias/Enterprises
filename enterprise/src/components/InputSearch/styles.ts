import styled from 'styled-components/native';

export const Touchable = styled.TouchableOpacity`
  border-width: 1px;
  border-color: #f3f3f3;
  width: 100%;
  height: 50px;
  border-radius: 25px;
  margin-top: -8px;

  align-items: center;
  flex-direction: row;
  padding: 0 16px;
`;

export const ContentInput = styled.View`
  width: 90%;
`;

export const Input = styled.TextInput`
  color: #a4a4b2;
  margin-left: 8px;
  font-family: 'Poppins-Regular';
  margin-top: 4px;
`;
