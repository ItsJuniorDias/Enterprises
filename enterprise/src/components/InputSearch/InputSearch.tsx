import React, { useRef } from 'react';
import { Image, TextInputProps } from 'react-native';
import { Touchable, ContentInput, Input } from './styles';
import search from '../../assets/search.png';
interface ISearch extends TextInputProps {
  title: string;
  name: string;
  callBackParent(value: any): void;
}

export const InputSearch = ({
  testID = 'inputSearch_testId',
  title,
  name,
  callBackParent,
  ...rest
}: ISearch) => {
  const inputElementRef = useRef<any>(null);

  const testIDs = useRef({
    input_search: 'inputSearch_testId',
    touchable: 'touchable_testID',
  }).current;

  return (
    <Touchable
      testID={testIDs.touchable}
      onPress={() => inputElementRef.current.focus()}
    >
      <Image source={search} />

      <ContentInput>
        <Input
          testID={testIDs.input_search}
          placeholder={title}
          placeholderTextColor="#A4A4B2"
          ref={inputElementRef}
          onChangeText={callBackParent}
          {...rest}
        />
      </ContentInput>
    </Touchable>
  );
};
