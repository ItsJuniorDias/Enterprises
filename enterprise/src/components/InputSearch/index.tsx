import React, { useRef } from 'react';
import { Image, TextInput, View } from 'react-native';

import { Touchable, ContentInput, Input } from './styles';

import search from '../../assets/search.png';

interface ISearch {
  title: string;
}

const InputSearch: React.FC<ISearch> = ({ title }) => {
  const inputElementRef = useRef<any>(null);

  return (
    <Touchable onPress={() => inputElementRef.current.focus()}>
      <Image source={search} />

      <ContentInput>
        <Input
          placeholder={title}
          placeholderTextColor="#A4A4B2"
          ref={inputElementRef}
        />
      </ContentInput>
    </Touchable>
  );
};

export default InputSearch;
