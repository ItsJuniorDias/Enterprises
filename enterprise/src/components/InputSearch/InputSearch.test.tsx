import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { InputSearch } from '../InputSearch/InputSearch';

describe('Behavior InputSearch', () => {
  const handleFilterValue = jest.fn();

  const screenRender = (
    <InputSearch
      title="Buscar por nome"
      name="filter"
      callBackParent={(value) => handleFilterValue(value)}
    />
  );

  it('render snapshot', () => {
    const result = render(screenRender).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('should press touchable', () => {
    const { getByTestId } = render(screenRender);

    const touchable = getByTestId('touchable_testID');

    fireEvent.press(touchable);

    expect(touchable).toBeTruthy();
  });
});
