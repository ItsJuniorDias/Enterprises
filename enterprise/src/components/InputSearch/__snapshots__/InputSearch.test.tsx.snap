// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Behavior InputSearch render snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#f3f3f3",
      "borderRadius": 25,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "marginTop": -8,
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 0,
      "width": "100%",
    }
  }
  testID="touchable_testID"
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/search.png",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "width": "90%",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Buscar por nome"
      placeholderTextColor="#A4A4B2"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "color": "#a4a4b2",
            "fontFamily": "Poppins-Regular",
            "marginLeft": 8,
            "marginTop": 4,
          },
        ]
      }
      testID="inputSearch_testId"
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;
